src/main/java/org/sonarlint/intellij/config/global/OrganizationSelector.java

```diff
@@ -0,0 +1,159 @@
+    orgList.setEnabled(false);
+    orgList.setEmptyText("Fetching list from SonarQube server..");
+    orgList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+    orgList.setCellRenderer(new ListRenderer());
+    orgList.addMouseListener(new MouseDoubleClickAdapter());
+    orgList.addListSelectionListener(e -> organizationSelected());
+
+    Convertor<Object, String> convertor = o -> {
+      RemoteOrganization org = (RemoteOrganization) o;
+      return org.getName() + " " + org.getKey();
+    };
+    new ListSpeedSearch(orgList, convertor);
+
+    panel.add(text, BorderLayout.NORTH);
+    JScrollPane scrollPane = new JScrollPane(orgList);
+    panel.add(scrollPane, BorderLayout.CENTER);
+    new UpdateTask().queue();
+    return panel;
+  }
+
+  private class MouseDoubleClickAdapter extends MouseAdapter {
+    @Override
+    public void mouseClicked(MouseEvent e) {
+      if (e.getClickCount() == 2 && orgList.getSelectedValue() != null) {
+        RemoteOrganization org = (RemoteOrganization) orgList.getSelectedValue();
+        selectedOrganizationKey = org.getKey();
+        setOKActionEnabled(true);
+        OrganizationSelector.this.close(0, true);
+      }
+    }
+  }
+
+  private void organizationSelected() {
+    RemoteOrganization org = (RemoteOrganization) orgList.getSelectedValue();
+    this.selectedOrganizationKey = org.getKey();
+    this.setOKActionEnabled(true);
+  }
+
+  private class UpdateTask extends Task.Backgroundable {
+    public UpdateTask() {
+      super(null, "Updating List of Organizations", true);
+    }
+
+    public void run(ProgressIndicator progress) {
+      try {
+        ServerConfiguration serverConfiguration = SonarLintUtils.getServerConfiguration(server);
+        WsHelperImpl impl = new WsHelperImpl();
+        List<RemoteOrganization> orgs = impl.listOrganizations(serverConfiguration, null);
+        ApplicationManager.getApplication().invokeLater(() -> setResults(orgs), ModalityState.stateForComponent(panel));
+      } catch (Exception e) {
+        ApplicationManager.getApplication().invokeLater(() -> {
+          Messages.showMessageDialog(panel, "Please check your server configuration: " + e.getMessage(),
+            "Failed to Download List of Organizations", Messages.getWarningIcon());
+          OrganizationSelector.super.close(0, false);
+        }, ModalityState.stateForComponent(panel));
+      }
+    }
+
+    public void setResults(List<RemoteOrganization> orgs) {
+      orgList.setListData(orgs.toArray());
+      orgList.setEnabled(true);
+      orgList.requestFocusInWindow();
+    }
+  }
+
+  public String getSelectedOrganizationKey() {
+    return selectedOrganizationKey;
+  }
+
+  private class ListRenderer extends ColoredListCellRenderer<RemoteOrganization> {
+    @Override protected void customizeCellRenderer(JList<? extends RemoteOrganization> list, @Nullable RemoteOrganization value, int index, boolean selected, boolean hasFocus) {
+      if (value == null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2583)